src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,104 @@
+{
+    TrajectoryElement_t* trajectory_elements;
+    unsigned int num_elements;
+} Trajectory_t;
+
+typedef struct FirmwareRobotPathParameters
+{
+    Polynomial2dOrder3_t path;
+    double t_start;
+    double t_end;
+    unsigned int num_segments;
+    double max_allowable_acceleration;
+    double max_allowable_speed;
+    double initial_speed;
+    double final_speed;
+} FirmwareRobotPathParameters_t;
+
+/**
+ * Returns a planned trajectory as a list of TrajectoryElement's.
+ *
+ * Key assumptions of this planner are:
+ *  - Trajectories are time-optimal assuming INFINITE JERK capability of the robot
+ *  - No speed on the trajectory is larger than the 'max_allowable_speed' parameter.
+ *  - No acceleration value between points on the trajectory can be larger than the
+ * 'max_allowable_acceleration' input parameter
+ *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION and at no
+ * point on the path can the sum of centripital and acceleration force be greather than
+ * 'max_allowable_acceleration'
+ *
+ *  Trajectory generation is done by assuming constant acceleration capability.
+ *      - The generator will assume maximum acceleration for the robot between each
+ * segment on the path. If using max acceleration breaches either the speed limit of the
+ * robot, or the centripital acceleration limit defind by the curvature and robot speed -
+ * the planner will assume the max acceleration that will remain bellow this limit.
+ *      - To ensure the robot is capable of deccelerating to reach ant sudden changes in
+ * the path, the trajectory is checked for backwards continuity
+ *
+ * @pre t_start < t_end
+ * @pre num_segments > 2
+ * @pre max_allowable_acceleration > 0
+ * @pre max_allowable_speed > 0
+ * @pre init_speed > 0
+ * @pre final_speed > 0
+ *
+ *  @param path_parameters [in] The data structure including important path parameters
+ * such as: *path - The 3D polynomial representing the geometric path
+ *
+ *              *t_start - The path parameter value indicating the beggining of the
+ * considered path *t_end - The path parameter value indicating the end of the considered
+ * path
+ *
+ *              *num_segments - The number of segments to discreteize the trajectory into.
+ *                          Must be greater than 2 segments.
+ *                          THE NUMBER OF SEGMENTS MUST BE UNDER
+ *                          TRAJECTORY_PLANNER_MAX_NUMER_SEGMENTS
+ *
+ *              *max_allowable_acceleration - The maximum acceleration allowed at any
+ * point along the trajectory. This factor limits the maximum delta-velocity and also the
+ * max speed around curves due to centripital acceleration [m/s^2]
+ *
+ *              *max_allowable_speed - The maximum speed allowable at any point along the
+ * path [m/s] *initial_speed - The initial speed at the start of the trajectory [m/s]
+ *
+ *              *final_speed - The final speed at the end of the trajectory [m/s]
```

Comment:
All of these parameters are part of the `struct` right? If so, we should document them in the struct definition instead of in a function jdoc.

Reply:
Moved comments to the struct definition